plugin/proto.go

```diff
@@ -0,0 +1,56 @@
+// Package plugin defines the protocol layer for communication between notation
+// and notation external plugin.
+package plugin
```

Comment:
Yes, we should but since notation-go will depend on this package not other way around I had to create it here. Once we update notation-go to depend on this pkg we can remove it from notation-go